x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java

```diff
@@ -671,7 +671,7 @@ public void putOrUpdateTransformStoredDoc(
                 ActionListener.wrap(
                     r -> listener.onResponse(SeqNoPrimaryTermAndIndex.fromIndexResponse(r)),
                     e -> listener.onFailure(
-                        new RuntimeException(
+                        new TransformConfigException(
```

Comment:
Are there any cases when the inability to persist stats is unrelated to transform config version conflict? I think it could happen but not sure if this was a real issue in the past.
If so, probably we should rename this exception class?

Reply:
Yeah I assumed so, like a node disconnect or something of that sort, but that is why I went with a more generic `TransformConfigException`.  Do you have any suggestions?

Reply:
I thought about sth like `TransformStatePersistenceException`.